src/op_mode/nhrp.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2023 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import tabulate
+import vyos.opmode
+
+from vyos.util import cmd
+from vyos.util import process_named_running
+
+
+def get_formatted_output(output_list: list) -> str:
+    """
+    Create formatted table for CLI output
+    :param output_list: list for API
+    :type output_list: list
+    :return: tabulate string
+    :rtype: str
+    """
+    status: dict = output_list.pop(0)
+    if 'Status' in status:
+        print(f"Status: {status['Status']}")
+    output: str = tabulate.tabulate(output_list, headers='keys',
+                                    numalign="left")
+    return output
+
+
+def get_formatted_list(output_string: str) -> list:
```

Comment:
First, I believe for such relatively obscure formats we should always include output samples in the code, or at least describe what the format is.

Second, the current JSON output is very annoying to work with for any client:

```
[{'Status': 'ok'},
 {'Interface': 'tun0', 'Type': 'local', 'Protocol-Address': '10.4.0.255/32', 'Alias-Address': '10.4.0.1', 'Flags': 'up'}, {'Interface': 'tun0', 'Type': 'local', 'Protocol-Address': '10.4.0.1/32', 'Flags': 'up'}, {'Interface': 'tun0', 'Type': 'local', 'Protocol-Address': '10.4.0.255/32', 'Alias-Address': '10.4.0.1', 'Flags': 'up'}, {'Interface': 'tun0', 'Type': 'local', 'Protocol-Address': '10.4.0.1/32', 'Flags': 'up'}]
```

The list contains objects of different types: the status object and a bunch of route objects, so any client code has to separate them, it cannot just loop through the routes.

At the very least we should do something like this:

```
{ 'status': 'Ok",
  'routes': [{'Interface': 'tun0', ...}, ...]
```

Reply:
done